### DISCUSSION TITLE
Keep Typed Value in HxSearchBox?

### DISCUSSION THREAD

**Discussion by chobo2** [havit/Havit.Blazor#679] [Q&A] [answered]
Hi

I noticed if I go to the example and type "John Smith" and choose one of the options everything in the box gets cleared out.

How can I stop that from happening?

https://havit.blazor.eu/components/HxAutosuggest

**Comment by chobo2** [comment]
I see "SelectAndClearTextQuery" and TextQuery I think i just need to change SelectAndClearTextQuery ? I am not sure What TextQuery does or if it is needed.

**Reply by hakenr** [reply]
Yes, `SelectAndClearTextQuery` can be set to preserve the text in the input after triggering the query.

**Comment by chobo2** [comment]
Also, why is TextQueryTriggered only "on enter" and not on "text changed" or blur?

It seems like I have to hit "Enter" to record the value they typed. 

Do I need to use the TextQueryChanged as well as the Triggered or one or the other?

**Reply by hakenr** [reply]
This is intentional, the `HxSearchBox` is meant to be used for search-scenarios where the user interaction ends with either selecting one of the items, triggering the search with `Enter` or abandoning the search (where the search-box gets cleared).

What is your usage scenario? Why do you want to preserve the input content without triggering the search?

**Reply by chobo2** [reply, marked as answer]
Well none of the controls really meet exactly my needs, so I got to sort of bend them a bit. I have 3 "boxes" on my page that all must have "suggestions" the first box is an autosuggest box as they must choose from the results that come back.

The other 2, I am making them as "searchbox" as I need to allow free text as they maybe looking for a very general result (ie they probably ignore the suggestions that come back) or they may take the suggestion. 

All 3 of the boxes are used to make one search.  IE at the bottom I have a "search button". That is why "enter" does not really work as they probably will tab to the next box. Then click the "search" 

This is what I came up with and it seems to work.
```
<HxSearchBox DataProvider="ProvideSearchResults"
			 Label="Employee"
			 TItem="string?"
			 ItemTitleSelector="i => i"
			 Placeholder="Search"
			  ItemSelectionBehavior="SearchBoxItemSelectionBehavior.SelectAndReplaceTextQueryWithItemTitle"
			 TextQueryChanged="QueryChanged"
			 TextQuery="@textQueryValue">
	<DefaultContentTemplate>
		<div class="p-2 text-muted">Search for employees by name...</div>
	</DefaultContentTemplate>
	<NotFoundTemplate>
		<div class="small py-2 px-3 text-muted"><HxIcon CssClass="me-2" Icon="BootstrapIcon.EmojiFrown" />Sorry, I did not find that...</div>
	</NotFoundTemplate>

<HxSearchBox DataProvider="ProvideSearchResults2"
			 Label="Employee2"
			 TItem="string?"
			 ItemTitleSelector="i => i"
			 Placeholder="Search"
			  ItemSelectionBehavior="SearchBoxItemSelectionBehavior.SelectAndReplaceTextQueryWithItemTitle"
			 TextQueryChanged="QueryChanged2"
			 TextQuery="@textQueryValue2">
	<DefaultContentTemplate>
		<div class="p-2 text-muted">Search for employees by name...</div>
	</DefaultContentTemplate>
	<NotFoundTemplate>
		<div class="small py-2 px-3 text-muted"><HxIcon CssClass="me-2" Icon="BootstrapIcon.EmojiFrown" />Sorry, I did not find that...</div>
	</NotFoundTemplate>


</HxSearchBox>
<HxButton Text="Search" OnClick="OnSearchClick" />
@code {
   private string? textQueryValue = null;
   private void QueryChanged(string text){
         textQueryValue = text;
   }

   private string? textQueryValue2 = null;
   private void QueryChanged2(string text){
         textQueryValue2 = text;
   }

   private void OnSearchClick(){
            myservice.doSearch(textQueryValue, textQueryValue2);
   }
}
```
